src/uu/ls/src/ls.rs

```diff
@@ -3542,3 +3542,15 @@ fn calculate_padding_collection(
 
     padding_collections
 }
+
+fn escape_name(name: &OsStr, style: &QuotingStyle) -> String {
```

Comment:
i am confused why these functions ?

Reply:
Implementing full support for non-unicode characters in `ls` requires handling lots of edge cases for how to handle structured output when it's not printable, so it will have to wait for a future PR. These functions wrap the new implementations to punt these edge cases into something closer to what we were doing previously, so that we don't have to make extensive changes to the rest of the code yet. Even with these wrappers, we're still most of the way there anyway, since control characters and escape sequences are now properly handled (the prior implementation threw away all non-ASCII control characters and all non-Unicode bytes in the *input*, these wrappers only throw away non-Unicode bytes in the *output*, which only `--literal` needs).

Reply:
ok, could you please document this into the code too? It feels weird creating this function for one call

Reply:
`escape_name` is used 5 times in this file, so I'm guessing you mean for `escape_dir_name`? I used that for symmetry with `escape_name`, if you prefer I can just move it inline.

Reply:
This is a change I did in #6559 and was discussed [here](https://github.com/uutils/coreutils/pull/6559#discussion_r1715974728)

Reply:
Sorry, i forgot

Reply:
No problem :+1: you have to deal with massive context switches for maintaining the coreutils, that's not an easy task.
(Thank you a lot for fulfilling this role btw, you do an amazing job :pray:).

Reply:
Easy with contributors like you folks ;)

Reply:
Ended up adding a FIXME comment to `escape_name`, and moving `escape_dir_name` inline with the same comment.